Validator: Added warnings and info messages for Mutation_Status column for mutations_extended validation

dionnezaal opened the pull request:
# What? Why?
When loading mutation data we encountered that when mutation files contain 'LOH' or 'None' ([allowed values for Mutation_Status in MAF format](https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification)) in the Mutation_Status where not loaded by the importer, but no warning was given to the user in the validation script.  Also, only Mutation_Status 'Somatic' and 'Germline' are used by cBioPortal and given a symbol in the mutation tab, but if values differ from these no warning is given to the user. 

## Changes proposed in this pull request
- Give warning to the user that line in mutation file will not be loaded when mutation file contains 'LOH' or 'None' in Mutation_Status column. 
- Give warning to the user when Mutation_Status column contains different values than 'Somatic' or 'Germline'.

<img width="1128" alt="screen shot 2018-01-09 at 12 30 09 pm" src="https://user-images.githubusercontent.com/1343609/34718941-e87d1b3a-f538-11e7-9ece-b8b94b771e50.png">

dionnezaal commented:
After discussion with @sandertan changed the mutation not loaded to a info message. Also changed warning when Mutation_Status column contains different values than 'Somatic' or 'Germline' to allow  'Wildtype' and all allowed Mutation_Status values from the MAF format.

sandertan commented on core/src/test/scripts/unit_tests_validate_data.py:
```diff
@@ -1349,6 +1349,55 @@ def test_validation_verification_status(self):
+        # We expect 5 info messages, 3 from not loaded mutations and 2 general lines
+        self.assertEqual(len(record_list), 5)
+        record_iterator = iter(record_list)
+        # Validation Status error
```
Please change this comment and the 2 below to something like "# Expecting info message"

sandertan commented on core/src/test/scripts/unit_tests_validate_data.py:
```diff
@@ -1349,6 +1349,55 @@ def test_validation_verification_status(self):
+        record_list = self.validate('mutations/data_mutations_ms.maf',
+                                    validateData.MutationsExtendedValidator,
+                                    extra_meta_fields={'swissprot_identifier': 'name'})
+        # we expect 1 error
```
warning

sandertan commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -1507,6 +1508,16 @@ def checkFilterAnnotation(self, value):
+    def checkMutationStatus(self, value):
+        """Check values in mutation status column."""
+        if value in ['LOH', 'None', 'Wildtype']:
+            self.logger.info('Mutation will not be loaded due to Mutation Status',
```
Can we make it `Mutation will not be loaded due to value in Mutation_Status`?

sandertan commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -1507,6 +1508,16 @@ def checkFilterAnnotation(self, value):
+            self.logger.info('Mutation will not be loaded due to Mutation Status',
+                                extra={'line_number': self.line_number, 'cause': value})
+        if value not in ['None', 'Germline', 'Somatic', 'LOH', 'Post-transcriptional modification', 'Unknown', 'Wildtype'] and value != '':
+            self.logger.warning('Mutation status is not in MAF format',
```
`Mutation_Status value is not in MAF format`

sandertan commented on core/src/test/scripts/unit_tests_validate_data.py:
```diff
@@ -1378,24 +1378,24 @@ def test_mutation_not_loaded_ms(self):
         self.assertEqual(record.cause, 'None')
-        self.assertEqual(record.getMessage(), "Mutation will not be loaded due to Mutation Status")
-        # Validation Status error
+        self.assertEqual(record.getMessage(), "Mutation will not be loaded due to value in Mutation Status")
```
Lets make this `Mutation_Status` for consistency.

dionnezaal commented:
Applied last requested change from @sandertan, also added update for documentation about the `Mutation_Status` to File-Formats.md, squashed all commits and rebased on master.

dionnezaal commented:
Rebased PR on master and solved merge conflict in docs/File-Formats.md
